### PR TITLE
Allow to set fcuid to log explicitly

### PR DESCRIPTION
**Pull request by khancyr**
this is mandatory for multi fcu configuration to select the right one to log

### CODE REVIEW

File: examples/config.sample [hunk 1]
```diff
@@ -53,6 +53,12 @@
 #       most verbose.
 #       Default:<info>
 #
+#   FcuId
```
Comment (lucasdemarchi) [inline examples/config.sample:56]: suggestion: LogSystemId, and move it near to "LogMode".

File: src/logendpoint.h [hunk 2]
```diff
@@ -44,6 +44,7 @@ struct LogOptions {
     MavDialect mavlink_dialect{MavDialect::Auto}; // conf "MavlinkDialect"
     unsigned long min_free_space;                 // conf "MinFreeSpace"
     unsigned long max_log_files;                  // conf "MaxLogFiles"
+    int fcu_id{-1};                                   // conf "LogMasterId"
```
Comment (lucasdemarchi) [inline src/logendpoint.h:47]: LogMasterId?

File: src/binlog.cpp [hunk 3]
```diff
@@ -101,10 +101,13 @@ int BinLog::write_msg(const struct buffer *buffer)
         source_component_id = msg->compid;
     }
 
-    /* set the expected system id to the first autopilot that we get a heartbeat from */
-    if (_target_system_id == -1 && msg_id == MAVLINK_MSG_ID_HEARTBEAT
-        && source_component_id == MAV_COMP_ID_AUTOPILOT1) {
-        _target_system_id = source_system_id;
+    /* set the expected system id to the first autopilot that we get a heartbeat from if fcu_id isn't settled */
+    if (_target_system_id == -1) {
```
Comment (lucasdemarchi) [inline src/binlog.cpp:105]: maybe easier to just set _target_system_id in the constructor looking at the config value we receive as parameter?

**Comment by lucasdemarchi** [comment]
Overall looks good, thanks. This should fix #163

File: src/logendpoint.h (hunk 2 above)
Comment (khancyr) [reply src/logendpoint.h:47]: leftover from some thinking on the name...

File: examples/config.sample (hunk 1 above)
Comment (khancyr) [reply examples/config.sample:56]: ok

File: src/binlog.cpp (hunk 3 above)
Comment (khancyr) [reply src/binlog.cpp]: done

**Comment by khancyr** [comment]
tested with ArduPilot SITL: 
test target id out of range
test SITL ID 1, mavlink-router target 2 : no log
test SITL ID 2, mavlink-router target 2 : log
test SITL ID 42, mavlink-router no target settle : lLogging target system_id=42

**Pull request merged**